modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/ScalaCaskServerCodegen.java

```diff
@@ -0,0 +1,965 @@
+     * {{{
+     * //conflict: @cask.get("/user/:username")
+     * def getUserByName(username: String, request: cask.Request) = ...
+     * }}}
+     * <p>
+     * and we introduce a new discriminator function to "manually" call those conflicts:
+     * {{{
+     *
+     * @cask.get("/user", subpath = true)
+     * def userRouteDescriminator(request: cask.Request) = {
+     * request.remainingPathSegments match {
+     * case Seq("logout") => logoutUser(request)
+     * case Seq("login") => loginUser(request)
+     * case Seq(param) => getUserByName(param, request)
+     * }
+     * }
+     * }}}
+     */
+    public static class OperationGroup {
+        List<CodegenOperation> operations = new ArrayList<>();
+        final String pathPrefix;
+        final String httpMethod;
+        final String caskAnnotation;
+        final String methodName;
+
+        // TODO - multiple operations may have the same query params, so we'll need to somehow merge them (and take the right type)
+        public boolean hasGroupQueryParams() {
+            return operations.stream().flatMap(op -> op.queryParams.stream()).count() > 0;
+        }
+
+        /**
+         * This is invoked from `scala-cask/apiRoutesQueryParamsTyped.mustache`
+         *
+         * @return the CodegenParameters
+         */
+        public List<CodegenParameter> getGroupQueryParams() {
+            List<CodegenParameter> list = operations.stream().flatMap(op -> op.queryParams.stream()).map(p -> {
+                        final CodegenParameter copy = p.copy();
+                        copy.vendorExtensions.put("x-default-value", defaultValue(p));
+                        copy.required = false; // all our query params are optional for our work-around as it's a super-set of a few different routes
+                        copy.dataType = asScalaDataType(copy, false, true, true);
+                        copy.defaultValue = defaultValue(copy);
+                        return copy;
+                    }
+            ).collect(Collectors.toList());
+
+            return list;
+        }
+
+        @Override
+        public String toString() {
+            List<String> ops = operations.stream().map(o -> o.path + "\n").collect(Collectors.toList());
+            return httpMethod + " " + pathPrefix + " w/ " + operations.size() + " operations:\n" + String.join("", ops);
+        }
+
+        public OperationGroup(String httpMethod, String pathPrefix) {
+            this.httpMethod = httpMethod;
+            this.pathPrefix = pathPrefix;
+            caskAnnotation = "@cask." + httpMethod.toLowerCase(Locale.ROOT);
+
+            String[] array = null;
+            try {
+                array = pathPrefix.split(File.separator, -1);
+            } catch (PatternSyntaxException exp) {
+                throw new RuntimeException("For some reason, splitting >" + pathPrefix + "< on >" + File.separator + "< threw " + exp, exp);
+            }
```

Comment:
looks like this line is throwing exception in windows?

what about using just `/` instead of File.separator, which is `\` in windows?

Reply:
good catch - done